File: Day2/calculator.go
package main

import (
	"fmt"
	"errors"
)

func perform_operation(op string, num1 float64, num2 float64) (float64, error) {
	switch op {
	case "+":
		return num1 + num2, nil
	case "-":
		return num1 - num2, nil
	case "*":
		return num1 * num2, nil
	case "/":
		if num2 == 0 {
			return 0, errors.New("Cannot divide by zero")
		}
		return num1 / num2, nil
	default:
		return 0, errors.New("Invalid operator")
	}
}

func main()  {
	fmt.Println("Enter the first number: ")
	var num1 float64
    fmt.Scanln(&num1)

	fmt.Println("Enter an operator (+, -, *, /): ")
	var op string
	fmt.Scanln(&op)

	fmt.Println("Enter the second number: ")
	var num2 float64
	fmt.Scanln(&num2)

	result, err := perform_operation(op,float64(num1),float64(num2))

	if err != nil {
		fmt.Println("Error: ", err)
	}else{
		fmt.Println("Result: ", result)
	}

	
}

File: 2_calculator/calculator.rb
def main
    puts "Enter the first number: "
    input1 = gets.chomp
  
    unless numeric?(input1)
      puts "That's not a valid number."
      return
    end
  
    num1 = input1.to_f
  
    puts "Enter an operator (+,-,*,/): "
    op = gets.chomp
  
    puts "Enter another number: "
    input2 = gets.chomp
  
    unless numeric?(input2)
      puts "That's not a valid number."
      return
    end
  
    num2 = input2.to_f
  
    result = perform_operation(op, num1, num2)
  
    if result == "Wrong Operator"
      puts result
    else
      puts "The result is: #{result}"
    end
  end
  
  def perform_operation(op, num1, num2)
    case op
    when "+"
      num1 + num2
    when "-"
      num1 - num2
    when "*"
      num1 * num2
    when "/"
      if num2 == 0
        "Cannot divide by zero"
      else
        num1 / num2
      end
    else
      "Wrong Operator"
    end
  end
  
  def numeric?(string)
    true if Float(string) rescue false
  end
  
main
  